### PR TITLE
Additional data cleanup for dob violations

### PR DESCRIPTION
**Pull request by chrishenry**
Resolves https://github.com/chrishenry/deltanyc-heatseek/issues/51

# Changelog

### Fix

* Additional data cleanup for dob violations. [Chris Henry]

### CODE REVIEW

File: data-imports/clean_utils.py [hunk 1]
```diff
@@ -67,14 +67,35 @@ def make_index(table, column):
...
+    UPDATE {table} SET {lot} = REPLACE({lot}, ' ', 0);
     UPDATE {table} SET bbl =
-            concat(trim({boro}), trim(LPAD({block}, 5, '0')), trim(LPAD({lot}, 4, '0')));
+            concat(trim({boro}), trim(LPAD({block}, 5, '0')), trim(LPAD({lot}, 4, '0')))
+            WHERE {boro} REGEXP '^[0-9]+$' AND {block} REGEXP '^[0-9]+$' AND {lot} REGEXP '^[0-9]+$';
     ALTER TABLE {table} ADD INDEX (bbl);
     '''.format(table=table, boro=boro, block=block, lot=lot)
 
+    if not column_exists(table, 'bbl'):
+        sql = "ALTER TABLE {table} ADD COLUMN bbl bigint(13) NULL DEFAULT NULL;".format(table=table, boro=boro, block=block, lot=lot) + sql
```
Comment (rdlester) [inline data-imports/clean_utils.py:95]: Super nit: minorly more readable to split this into multiple lines.

**Review by rdlester — APPROVED** [review]

File: data-imports/clean_utils.py (hunk 1 above)
Comment (rdlester) [reply data-imports/clean_utils.py:95]: Not like we're going after readability here so w/e

File: data-imports/clean_utils.py [hunk 2]
```diff
@@ -67,14 +67,35 @@ def make_index(table, column):
...
+
+    return (result.rowcount == 1)
+
 def clean_bbl(table, boro, block, lot):
-    return '''
-    ALTER TABLE {table} ADD COLUMN bbl bigint(13) NULL DEFAULT NULL;
+    sql = '''
+    UPDATE {table} SET {block} = REPLACE({block}, ' ', 0);
+    UPDATE {table} SET {block} = REPLACE({block}, '`', '');
+    UPDATE {table} SET {lot} = REPLACE({lot}, ' ', 0);
```
Comment (rdlester) [inline data-imports/clean_utils.py:87]: It's late for this comment, but: maybe best to just preg_replace all non-digit characters?

Comment (chrishenry) [reply data-imports/clean_utils.py:87]: Ya, good call. Tracked here; https://github.com/chrishenry/deltanyc-heatseek/issues/77

**Pull request merged**